core/parentchain/light-client/src/concurrent_access.rs

```diff
@@ -46,12 +48,14 @@ lazy_static! {
 /// either a mutating, or a non-mutating function on the validator.
 /// The reason we have this additional wrapper around `SealedIO`, is that we need
 /// to guard against concurrent access by using RWLocks (which `SealedIO` does not do).
-pub trait ValidatorAccess<ParentchainBlock>
+pub trait ValidatorAccess<ParentchainBlock, OCallApi: EnclaveOnChainOCallApi>
```

Comment:
(Can't put a review comment in an unchanged line, so I put it here): Remove the `lazy_static` `VALIDATOR_LOCK`, as it is no longer needed.